low-level/sensors/impl/platforms/realstor/realstor_disk_sensor.py

```diff
@@ -271,6 +271,17 @@ def _rss_check_disks_presence(self):
             #raise alert for added drive
             self._rss_raise_disk_alert(self.rssencl.FRU_INSERTION, disk_info)
 
+            # Update health status for inserted disk in memfault cache,
+            # to raise fault alert after insertion if inserted disk status is not OK.
+            if disk_info["health"] != "OK":
+                for id_fault, cached_fault in enumerate(self.rssencl.memcache_faults):
+                    #fetch disk slot from component_id present in memcache_faults.
+                    component_id = cached_fault["component-id"]
+                    if component_id.startswith('Disk 0'):
+                        disk_id = int(cached_fault["component-id"].split()[1].split('.')[1])
```

Comment:
There is an assumption about the format which may result into exception. We should plan to handle so that it does not break the code. 

```
for id_fault, cached_fault in enumerate(self.rssencl.memcache_faults):
          #fetch disk slot from component_id present in memcache_faults.
        try:  
                component_id = cached_fault["component-id"]
                if component_id.startswith('Disk 0'):
                       disk_id = int(cached_fault["component-id"].split()[1].split('.')[1])
                       if disk_id == slot:
                            self.rssencl.memcache_faults[id_fault]['health'] = "OK"

        except:
                log_error(....)

Reply:
agree, @satish-seagate plz address it. Also in exception, log it & proceed (basically ignore this condition then)

Reply:
Done